testng-junit-migrator/src/test/java/tech/picnic/errorprone/testngjunit/TestNGScannerTest.java

```diff
@@ -0,0 +1,231 @@
+      TestNGScanner scanner = new TestNGScanner(state);
+      ImmutableMap<ClassTree, TestNGMetadata> classMetaData =
+          scanner.collectMetadataForEachClass(tree);
+
+      classMetaData.forEach(
+          (classTree, metaData) -> {
+            metaData
+                .getClassLevelAnnotationMetadata()
+                .ifPresent(
+                    annotation ->
+                        state.reportMatch(
+                            buildDescription(annotation.getAnnotationTree())
+                                .setMessage(
+                                    createAnnotationDiagnosticMessage(classTree, annotation, state))
+                                .build()));
+
+            metaData
+                .getDataProviderMetadata()
+                .values()
+                .forEach(
+                    dataProvider -> {
+                      state.reportMatch(
+                          buildDescription(dataProvider.getMethodTree())
+                              .setMessage(
+                                  String.format(
+                                      "class: %s dataProvider: %s",
+                                      classTree.getSimpleName(), dataProvider.getName()))
+                              .build());
+                    });
+
+            metaData
+                .getSetupMethods()
+                .forEach(
+                    (method, setupMethodType) ->
+                        state.reportMatch(
+                            buildDescription(method)
+                                .setMessage(
+                                    String.format(
+                                        "class: %s setupMethod: %s",
+                                        classTree.getSimpleName(), setupMethodType))
+                                .build()));
+
+            classTree.getMembers().stream()
+                .filter(MethodTree.class::isInstance)
+                .map(MethodTree.class::cast)
+                .map(metaData::getAnnotation)
+                .flatMap(Optional::stream)
+                .filter(not(isEqual(metaData.getClassLevelAnnotationMetadata().orElse(null))))
+                .forEach(
+                    annotation -> {
+                      state.reportMatch(
+                          buildDescription(annotation.getAnnotationTree())
+                              .setMessage(
+                                  createAnnotationDiagnosticMessage(classTree, annotation, state))
+                              .build());
+                    });
+          });
+
+      return Description.NO_MATCH;
+    }
+
+    private static String createAnnotationDiagnosticMessage(
+        ClassTree classTree,
+        TestNGMetadata.AnnotationMetadata annotationMetadata,
+        VisitorState state) {
+      return String.format(
+          "class: %s arguments: { %s }",
+          classTree.getSimpleName(),
+          annotationMetadata.getArguments().entrySet().stream()
```

Comment:
I'm wondering if there is a simpler way of just printing the data. Tried this:
```
      return String.format(
          classTree.getSimpleName(), annotationMetadata.getArguments().toString());
```
Results in:
 
> /A.java:18: error: [TestChecker] Class: A arguments: { {} }
  @Test
  ^
    (see https://errorprone.info/bugpattern/TestChecker)
/A.java:22: error: [TestChecker] Class: A arguments: { {description="foo"} }
  @Test(description = "foo")
  ^
    (see https://errorprone.info/bugpattern/TestChecker)
/A.java:26: error: [TestChecker] Class: A arguments: { {priority=1, description="foo"} }
  @Test(priority = 1, description = "foo")
  ^
    (see https://errorprone.info/bugpattern/TestChecker)
/A.java:30: error: [TestChecker] Class: A arguments: { {dataProvider="dataProviderTestCases"} }
  @Test(dataProvider = "dataProviderTestCases")
  ^
    (see https://errorprone.info/bugpattern/TestChecker)
expected to be true

Not that bad right? I propose to use that :).